app/assets/javascripts/components/stories/StorySearch.js

```diff
@@ -0,0 +1,71 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import ReactDOM from 'react-dom';
+import { search } from "actions/story";
+import { connect } from "react-redux";
+
+const searchDiv = document.getElementById('portal-search-input');
+
+export class StorySearch extends Component {
+  constructor(props) {
+    super(props);
+
+    const { projectId, search } = props;
+
+    this.state = {
+      keyWord: '',
+      searchedStories: [],
+      projectId,
+      search
```

Comment:
I would suggest to you tu use Hooks since this project has React `16.8.0`:
```js
import React, { useState } from 'react'

const StorySearch = (props) => {
  const [keyWord, setKeyWord] = useState('');
  const [searchedStories, setSearchedStories] = useState([]);
  const [projectId, setProjectId] = useState(props.projectId);
  const [search, setSearch] = useState(props.search);

  const handleChange = e => setKeyWord(e.target.value);

  const handleSearch = async (e) => {
    e.preventDefault()

    search(keyWord, projectId)
  }

  return (
    <PortalSearchInput> //
      <form 
        action="#" 
        acceptCharset="UTF-8" 
        method="post"
        onSubmit={handleSearch}
      >
        <div className="form-group">
          <input 
            type="text" 
            onChange={handleChange} 
            placeholder="Search" 
            className="form-control input-sm"
            value={keyWord}
          />
        </div>
      </form>
    </PortalSearchInput>
  );
}
```

Then I realize these three states are unnecessary, you can use directly from `props`.